src/server/controller/cycleData/links/visitCycleLinks/worker.ts

```diff
@@ -42,12 +42,16 @@ export default async (job: VisitCycleLinksJob): Promise<void> => {
 
     const linkVisits = await visitLinks(mergedLinksToVisit)
 
-    await LinkRepository.upsertMany({
+    const updatedLinks = await LinkRepository.upsertMany({
       assessment,
       cycle,
       linkVisits,
     })
 
+    const excludedIds = updatedLinks.map((link) => link.id)
+
+    await LinkRepository.deleteMany({ assessment, cycle, excludeApproved: true, excludedIds })
```

Comment:
Do we want to remove links regardless if they are approved? if so, when they are deleted and added again later, they would need to be approved again.

Reply:
If we don't want to delete them then I think the best approach would be to update the `locations` field to an empty object. Then in the UI we show the links that are at least in one location. That would be more consistent with the current approach. What do you think?